lib/elixir/lib/calendar/duration.ex

```diff
@@ -331,4 +333,100 @@ defmodule Duration do
         raise ArgumentError, ~s/failed to parse duration "#{string}". reason: #{inspect(reason)}/
     end
   end
+
+  @doc """
+  Converts the given duration to [ISO 8601-2:2019](https://en.wikipedia.org/wiki/ISO_8601) format.
+
+  Note this function implements the *extension* of ISO 8601:2019. This extensions allows weeks to
+  appear in any case, which is not the case in the base standard. `"P3M3W"` is not allowed in
+  ISO 8601-1, but it is in ISO 8601-2. Thus, any duration can be output by this function.
+
+  ## Examples
+
+      iex> Duration.new!(year: 3) |> Duration.to_iso8601()
+      "P3Y"
+      iex> Duration.new!(day: 40, hour: 12, minute: 42, second: 12) |> Duration.to_iso8601()
+      "P40DT12H42M12S"
+      iex> Duration.new!(second: 30) |> Duration.to_iso8601()
+      "PT30S"
+      iex> Duration.new!(second: 30) |> Duration.multiply(-1) |> Duration.to_iso8601()
+      "PT-30S"
+
+      iex> Duration.new!([]) |> Duration.to_iso8601()
+      "PT0S"
+
+      iex> Duration.new!(second: 1, microsecond: {2_200, 3}) |> Duration.to_iso8601()
+      "PT1.002S"
+      iex> %Duration{second: 1, microsecond: {-1_200_000, 4}} |> Duration.to_iso8601()
+      "PT-0.2000S"
```

Comment:
Personally, i'd prefer reading the examples as:

```
Duration.to_iso8601(%Duration{second: 1, microsecond: {-1_200_000, 4}})
```

and wouldn't even include the example with `Duration.multiply/2`, but feel free to keep it as is of course

Reply:
Yep, fixed in https://github.com/elixir-lang/elixir/pull/13604/commits/c126b7c99a58b3d834280a7bba50d0215353cba1

It's more coherent with the rest of the examples, and `Duration.t()` is not an opaque type after all.